src/v/model/fundamental.h

```diff
@@ -53,6 +54,24 @@ inline constexpr offset prev_offset(offset o) {
     return o - offset{1};
 }
 
+inline constexpr offset operator+(offset o, offset_delta d) {
+    if (o >= offset{0}) {
+        if (d() <= offset::max() - o) {
+            return offset{o() + d()};
+        } else {
+            return offset::max();
+        }
+    } else {
+        if (d() >= offset::min() - o) {
+            return offset{o() + d()};
+        } else {
+            return offset::min();
+        }
+    }
+}
+
+inline constexpr offset operator-(offset o, offset_delta d) { return o + (-d); }
```

Comment:
@andijcr given the recent issue with named_type strictness (with the PR you had to address it), are these overloads scary?

Reply:
I'm not familiar...do you have a pointer?

Reply:
https://github.com/redpanda-data/redpanda/pull/15170 sorry i missed the tag.
i need to have a look, i think we already have some overloads for these operators and types...

Reply:
Ah, got it thanks. That looks like a good change 👍 

For context on this diff: `kafka::offset_delta` is new. the idea is to distinguish from `model::offset_delta` where the operator overloads do conversion back/forth between `kafka::offset` & `model::offset`.

as for the overload resolution itself, I think we're fine because the desired one is more specific? like this: 
https://godbolt.org/z/EfG7Yx5vc